src/simple-modal-holder.component.ts

```diff
@@ -0,0 +1,130 @@
+import { Component, ComponentFactoryResolver, Type, ViewChild, ViewContainerRef } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+
+import { SimpleModalWrapperComponent } from './simple-modal-wrapper.component';
+import { SimpleModalComponent } from './simple-modal.component';
+import { SimpleModalOptions } from './simple-modal-options';
+
+@Component({
+  selector: 'simple-modal-holder',
+  template: '<ng-template #viewContainer></ng-template>',
+})
+export class SimpleModalHolderComponent {
+
+  /**
+   * Target viewContainer to insert modals
+   */
+  @ViewChild('viewContainer', {read: ViewContainerRef}) public viewContainer: ViewContainerRef;
+
+  /**
+   * modal collection, maintained by addModal and removeModal
+   * @type {Array<SimpleModalComponent> }
+   */
+  modals: Array<SimpleModalComponent<any, any>> = [];
+
+  /**
+   * Constructor
+   * @param {ComponentFactoryResolver} resolver
+   */
+  constructor(private resolver: ComponentFactoryResolver) {}
+
+  /**
+   * Configures then adds modal to the modals array, and populates with data passed in
+   * @param {Type<SimpleModalComponent>} component
+   * @param {object?} data
+   * @param {SimpleModalOptions?} options
+   * @return {Observable<*>}
+   */
+  addModal<T, T1>(component: Type<SimpleModalComponent<T, T1>>, data?: T, options?: SimpleModalOptions): Observable<T1> {
+    options = options || <SimpleModalOptions>{};
+
+    const factory = this.resolver.resolveComponentFactory(SimpleModalWrapperComponent);
+    const componentRef = this.viewContainer.createComponent(factory, options.index);
+    const modalWrapper: SimpleModalWrapperComponent = <SimpleModalWrapperComponent> componentRef.instance;
+    const _component: SimpleModalComponent<T, T1> =  modalWrapper.addComponent(component);
+
+    if (typeof(options.index) !== 'undefined') {
+      this.modals.splice(options.index, 0, _component);
+    } else {
+      this.modals.push(_component);
+    }
+    setTimeout(() => {
+      modalWrapper.wrapper.nativeElement.classList.add('show');
+      modalWrapper.wrapper.nativeElement.classList.add('in');
+    });
+    if (options.autoCloseTimeout) {
+      setTimeout(() => {
+        this.removeModal(_component);
+      }, options.autoCloseTimeout);
+    }
+    if (options.closeByClickingOutside) {
+      modalWrapper.onClickOutsideModalContent( () => {
+        this.removeModal(modalWrapper.content);
+      });
+    }
+    if (options.backdropColor) {
```

Comment:
Isn't this option going to be removed? If so, needs to be removed from the `simple-modal-options.ts` as well.

Reply:
yeah i think i'll create a second issue / pr for that and open up discussion to enhance the options maybe?

Reply:
I must reinforce that this is just my personal preference and opinion.

In this case, I'm against any type of style control via options. I think the style belong to the css, where it can be better integrated with the general styleguide of each individual project without the need to change a hex color, for example, in the javascript if it changes in the css. What if I want a background color with transparency of `0.54`? Maybe there will have to be another option to use in conjunction with the backdropColor, wich adds unecessary complexity to the code, when it's easily done in plain css.

It's better to maintain where it belongs. I can provide an initial css that is easy to understand and personalize. 

I think the options should be related to modal behavior only, such as closing by escape key, click outside, max number of modal open at the same time, etc.. 

And again, just my preference. If theres is a better argument to why we should keep it in the `js`, let's do it.